### ISSUE TITLE
authorizationCodeGrant is not a function

### ISSUE THREAD

**Issue by Ronnyrr** [thelinmichael/spotify-web-api-node#94]
Hi there.

I'm trying to authorize my Spotify account using you're example code as shown below: 

```
var credentials = {
  clientId : 'someClientId',
  clientSecret : 'someClientSecret',
  redirectUri : 'http://www.michaelthelin.se/test-callback'
};

var spotifyApi = new SpotifyWebApi(credentials);

// The code that's returned as a query parameter to the redirect URI
var code = 'MQCbtKe23z7YzzS44KzZzZgjQa621hgSzHN';

// Retrieve an access token and a refresh token
spotifyApi.authorizationCodeGrant(code)
  .then(function(data) {
    console.log('The token expires in ' + data.body['expires_in']);
    console.log('The access token is ' + data.body['access_token']);
    console.log('The refresh token is ' + data.body['refresh_token']);

    // Set the access token on the API object to use it in later calls
    spotifyApi.setAccessToken(data.body['access_token']);
    spotifyApi.setRefreshToken(data.body['refresh_token']);
  }, function(err) {
    console.log('Something went wrong!', err);
  });
```

But I can't seem to pass the **authorizationCodeGrant** part.

```
    this.spotifyApi = new SpotifyWebApi({
      clientId : '1234567',
      clientSecret : '8901234',
      redirectUri : 'http://localhost:6001/cv'
    });

    const scopes = ['user-read-private', 'user-read-email'];
    const state = '34fFs29kd09';

    const authorizeURL = this.spotifyApi.createAuthorizeURL(scopes, state);
    console.info(authorizeURL);

    this.spotifyApi.authorizationCodeGrant(code)
      .then(function(data) {
        console.log('The token expires in ' + data.body['expires_in']);
        console.log('The access token is ' + data.body['access_token']);
        console.log('The refresh token is ' + data.body['refresh_token']);

        // Set the access token on the API object to use it in later calls
        this.spotifyApi.setAccessToken(data.body['access_token']);
        this.spotifyApi.setRefreshToken(data.body['refresh_token']);
      }, function(err) {
        console.log('Something went wrong!', err);
      });
```

The URL given made my authorize and I've accepted. From there I'd get a new extremely long code:
`12345678901§2345678901234567890&state=34fFs29kd09`

Of course my code is different, but I don't know exactly if I'll have to use this code with or without the state parameter to the authorizationCodeGrant function. 

`this.spotifyApi.authorizationCodeGrant is not a function`

As my title suggests I can't seem to use the function authorizationCodeGrant as it says it does not exists. Any idea whats up?

**Comment by JMPerez** [comment]
@Ronnyrr are you running that code on the browser or on node?

**Comment by Ronnyrr** [comment]
I'm running it in a React app on componentDidMount(). So browser I guess?

**Comment by JMPerez** [comment]
Yes, that would be browser. For the [authorization code flow](https://developer.spotify.com/web-api/authorization-guide/#authorization-code-flow) you make the request server side, since you are sending your client secret and this needs to be hidden for security reasons. That's why the spotify-web-api-node library only exposes this function when using the code from node.

This is also explained on https://github.com/thelinmichael/spotify-web-api-node/issues/79 and https://github.com/thelinmichael/spotify-web-api-node#authorization

**Comment by Ronnyrr** [comment]
I really don't get what I'm doing wrong.
I've made the whole request serverside trough NodeJS but keep getting error messages like:

`Something went wrong on auth! { [WebapiError: Bad Request] name: 'WebapiError', message: 'Bad Request', statusCode: 400 }`

and 

`Something went wrong on playlists! { [WebapiError: Unauthorized] name: 'WebapiError', message: 'Unauthorized', statusCode: 401 }`

This afternoon I got it working once, but now the access token is expired or something I guess? But when I'll run the refresh function you posted in the authorization section I'll get:

`Could not refresh access token { [WebapiError: Bad Request] name: 'WebapiError', message: 'Bad Request', statusCode: 400 }`

How can I debug this? What can I do?
